vizro-core/examples/visual-vocabulary/pages/examples/diverging_stacked_bar.py

```diff
@@ -0,0 +1,122 @@
+from typing import Dict, List, Optional
+
+import pandas as pd
+import plotly.graph_objects as go
+import vizro.models as vm
+from vizro import Vizro
+from vizro.models.types import capture
+
+pastries = pd.DataFrame(
+    {
+        "pastry": [
+            "Scones",
+            "Bagels",
+            "Muffins",
+            "Cakes",
+            "Donuts",
+            "Cookies",
+            "Croissants",
+            "Eclairs",
+            "Brownies",
+            "Tarts",
+            "Macarons",
+            "Pies",
+        ],
+        "Profit Ratio": [-0.10, -0.15, -0.05, 0.10, 0.05, 0.20, 0.15, -0.08, 0.08, -0.12, 0.02, -0.07],
+        "Strongly Disagree": [20, 30, 10, 5, 15, 5, 10, 25, 8, 20, 5, 10],
+        "Disagree": [30, 25, 20, 10, 20, 10, 15, 30, 12, 30, 10, 15],
+        "Agree": [30, 25, 40, 40, 45, 40, 40, 25, 40, 30, 45, 35],
+        "Strongly Agree": [20, 20, 30, 45, 20, 45, 35, 20, 40, 20, 40, 40],
+    }
+)
+
+
+@capture("graph")
+def diverging_stacked_bar(
+    data_frame,
+    y: str,
+    category_pos: List[str],
+    category_neg: List[str],
+    color_discrete_map: Optional[Dict[str, str]] = None,
+) -> go.Figure:
+    """Creates a horizontal diverging stacked bar chart (with positive and negative values only) using Plotly's go.Bar.
+
+    This type of chart is a variant of the standard stacked bar chart, with bars aligned on a central baseline to
+    show both positive and negative values. Each bar is segmented to represent different categories.
+
+    This function is not suitable for diverging stacked bar charts that include a neutral category.
+
+    Inspired by: https://community.plotly.com/t/need-help-in-making-diverging-stacked-bar-charts/34023
+
+    Args:
+       data_frame (pd.DataFrame): The data frame for the chart.
+       y (str): The name of the categorical column in the data frame to be used for the y-axis (categories)
+       category_pos (List[str]): List of column names in the data frame representing positive values. Columns should be
+            ordered from least to most positive.
+       category_neg (List[str]): List of column names in the DataFrame representing negative values. Columns should be
+            ordered from least to most negative.
+       color_discrete_map: Optional[Dict[str, str]]: A dictionary mapping category names to color strings.
+
+    Returns:
+       go.Figure: A Plotly Figure object representing the horizontal diverging stacked bar chart.
+    """
```

Comment:
I reckon this is a very large docstring 😄 Shall we always add these docstrings to the code examples? Or would it suffice to only add these in the `custom_charts.py` ? The difference is that it would either appear or not in the code examples on the right.

Reply:
I really think we should leave docstrings in the examples.

Reply:
Yes, I agree. I think users also don't actually access `custom_charts.py` via the UI but only the code copy clipboard, so it probably needs to be added to both right now 👍